Build failed on SSLLocks.cpp on Centos 7

### Description

I have all dependencies satisfied. When building I get this error:

```
[ 10%] Building CXX object src/CMakeFiles/shared.dir/SSLLocks.cpp.o
/opt/sources/murmur/mumble-1.5.634/src/SSLLocks.cpp: In static member function ‘static void SSLLocks::initialize()’:
/opt/sources/murmur/mumble-1.5.634/src/SSLLocks.cpp:50:40: error: conversion to ‘unsigned int’ from ‘int’ may change the sign of the result [-Werror=sign-conversion]
   50 |  unsigned int nlocks = CRYPTO_num_locks();
      |                        ~~~~~~~~~~~~~~~~^~
cc1plus: all warnings being treated as errors
make[2]: *** [src/CMakeFiles/shared.dir/SSLLocks.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/shared.dir/all] Error 2
make: *** [all] Error 2
```


### Steps to reproduce

`cmake3 -Dclient=OFF ..`

```
-- The C compiler identification is GNU 9.3.1
-- The CXX compiler identification is GNU 9.3.1
-- Check for working C compiler: /opt/rh/devtoolset-9/root/usr/bin/cc
-- Check for working C compiler: /opt/rh/devtoolset-9/root/usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/rh/devtoolset-9/root/usr/bin/c++
-- Check for working CXX compiler: /opt/rh/devtoolset-9/root/usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python interpreter version 3.7.12
-- Performing Test COMPILER_HAS_MBIG_OBJ
-- Performing Test COMPILER_HAS_MBIG_OBJ - Failed
-- Using (among others) the following compiler flags: -Werror;-Wall;-Wpedantic;-Wextra;-Wconversion;-Wsign-conversion;-fsigned-char;-O3
-- ##################################################
-- Mumble version:              1.5.0
-- Architecture:                x64
-- Build type:                  Release
-- Using LTO in release builds: ON
-- ##################################################
-- Qt5 component found: Core | Version: 5.9.7
-- Qt5 component found: Network | Version: 5.9.7
-- Qt5 component found: Xml | Version: 5.9.7
-- OpenSSL component found: Crypto | Version: 1.0.2k
-- OpenSSL component found: SSL | Version: 1.0.2k
-- Protobuf found | Version: 3.19.1
-- Found Threads: TRUE
-- TRACY_ENABLE: OFF
-- TRACY_ON_DEMAND: ON
-- TRACY_CALLSTACK: OFF
-- TRACY_NO_CALLSTACK: OFF
-- TRACY_NO_CALLSTACK_INLINES: OFF
-- TRACY_ONLY_LOCALHOST: OFF
-- TRACY_NO_BROADCAST: OFF
-- TRACY_ONLY_IPV4: OFF
-- TRACY_NO_CODE_TRANSFER: OFF
-- TRACY_NO_CONTEXT_SWITCH: OFF
-- TRACY_NO_EXIT: OFF
-- TRACY_NO_SAMPLING: OFF
-- TRACY_NO_VERIFY: OFF
-- TRACY_NO_VSYNC_CAPTURE: OFF
-- TRACY_NO_FRAME_IMAGE: OFF
-- TRACY_NO_SYSTEM_TRACING: OFF
-- TRACY_DELAYED_INIT: OFF
-- TRACY_MANUAL_LIFETIME: OFF
-- TRACY_FIBERS: OFF
-- TRACY_NO_CRASH_HANDLER: OFF
-- TRACY_TIMER_FALLBACK: OFF
-- Disabling warnings for targets: TracyClient
-- Tracy: OFF
-- Found Git: /usr/bin/git (found version "1.8.3.1")
-- Qt5 component found: Sql | Version: 5.9.7
-- File: /opt/sources/murmur/mumble-1.5.634/build/src/murmur/murmur_autogen
-- Qt5 component found: DBus | Version: 5.9.7
-- Ice component found: Ice | Version: 3.7.10
-- Ice component found: IceSSL | Version: 3.7.10
-- Ice optional component found: IceDiscovery | Version: 3.7.10
-- Ice optional component found: IceLocatorDiscovery | Version: 3.7.10
-- Configuring done
-- Generating done
-- Build files have been written to: /opt/sources/murmur/mumble-1.5.634/build

```

`make`

```
Scanning dependencies of target TracyClient
[  1%] Building CXX object src/tracy/CMakeFiles/TracyClient.dir/public/TracyClient.cpp.o
[  2%] Linking CXX static library libTracyClient.a
[  2%] Built target TracyClient
Scanning dependencies of target generate_license_header
[  4%] Generating licenses.h
[  4%] Built target generate_license_header
Scanning dependencies of target shared_autogen
[  5%] Running cpp protocol buffer compiler on /opt/sources/murmur/mumble-1.5.634/src/MumbleUDP.proto
[  6%] Generating FFDHETable.h
[  8%] Running cpp protocol buffer compiler on /opt/sources/murmur/mumble-1.5.634/src/Mumble.proto
[  9%] Automatic MOC for target shared
[  9%] Built target shared_autogen
Scanning dependencies of target shared
[ 10%] Building CXX object src/CMakeFiles/shared.dir/shared_autogen/mocs_compilation.cpp.o
[ 12%] Building CXX object src/CMakeFiles/shared.dir/Mumble.pb.cc.o
[ 13%] Building CXX object src/CMakeFiles/shared.dir/MumbleUDP.pb.cc.o
[ 15%] Building CXX object src/CMakeFiles/shared.dir/Ban.cpp.o
[ 16%] Building CXX object src/CMakeFiles/shared.dir/EnvUtils.cpp.o
[ 17%] Building CXX object src/CMakeFiles/shared.dir/FFDHE.cpp.o
[ 19%] Building CXX object src/CMakeFiles/shared.dir/HostAddress.cpp.o
[ 20%] Building CXX object src/CMakeFiles/shared.dir/HTMLFilter.cpp.o
[ 21%] Building CXX object src/CMakeFiles/shared.dir/License.cpp.o
[ 23%] Building CXX object src/CMakeFiles/shared.dir/LogEmitter.cpp.o
[ 24%] Building CXX object src/CMakeFiles/shared.dir/MumbleProtocol.cpp.o
[ 26%] Building CXX object src/CMakeFiles/shared.dir/OSInfo.cpp.o
[ 27%] Building CXX object src/CMakeFiles/shared.dir/PasswordGenerator.cpp.o
[ 28%] Building CXX object src/CMakeFiles/shared.dir/PlatformCheck.cpp.o
[ 30%] Building CXX object src/CMakeFiles/shared.dir/QtUtils.cpp.o
[ 31%] Building CXX object src/CMakeFiles/shared.dir/ProcessResolver.cpp.o
[ 32%] Building CXX object src/CMakeFiles/shared.dir/ProtoUtils.cpp.o
[ 34%] Building CXX object src/CMakeFiles/shared.dir/SelfSignedCertificate.cpp.o
[ 35%] Building CXX object src/CMakeFiles/shared.dir/ServerAddress.cpp.o
[ 36%] Building CXX object src/CMakeFiles/shared.dir/ServerResolver.cpp.o
[ 38%] Building CXX object src/CMakeFiles/shared.dir/ServerResolverRecord.cpp.o
[ 39%] Building CXX object src/CMakeFiles/shared.dir/SSL.cpp.o
[ 41%] Building CXX object src/CMakeFiles/shared.dir/SSLLocks.cpp.o
/opt/sources/murmur/mumble-1.5.634/src/SSLLocks.cpp: In static member function ‘static void SSLLocks::initialize()’:
/opt/sources/murmur/mumble-1.5.634/src/SSLLocks.cpp:50:40: error: conversion to ‘unsigned int’ from ‘int’ may change the sign of the result [-Werror=sign-conversion]
   50 |  unsigned int nlocks = CRYPTO_num_locks();
      |                        ~~~~~~~~~~~~~~~~^~
cc1plus: all warnings being treated as errors
make[2]: *** [src/CMakeFiles/shared.dir/SSLLocks.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/shared.dir/all] Error 2
make: *** [all] Error 2
```


### Mumble version

1.5.634

### Mumble component

Server

### OS

Linux

### Reproducible?

Yes

### Additional information

_No response_

### Relevant log output

_No response_

### Screenshots

_No response_

--- a/src/SSLLocks.cpp
+++ b/src/SSLLocks.cpp
@@ -10,6 +10,8 @@
 
 #include <openssl/crypto.h>
 
+#include <cassert>
+
 static QMutex **locks = nullptr;
 
 void locking_callback(int mode, int type, const char *, int) {
@@ -47,9 +49,10 @@ unsigned long id_callback() {
 }
 
 void SSLLocks::initialize() {
-	unsigned int nlocks = CRYPTO_num_locks();
+	int nlocks = CRYPTO_num_locks();
+	assert(nlocks >= 0);
 
-	locks = reinterpret_cast< QMutex ** >(calloc(nlocks, sizeof(void *)));
+	locks = reinterpret_cast< QMutex ** >(calloc(static_cast< std::size_t >(nlocks), sizeof(void *)));
 	if (!locks) {
 		qFatal("SSLLocks: unable to allocate locks array");
 
@@ -60,7 +63,7 @@ void SSLLocks::initialize() {
 		exit(1);
 	}
 
-	for (unsigned int i = 0; i < nlocks; i++) {
+	for (unsigned int i = 0; i < static_cast< std::size_t >(nlocks); i++) {
 		locks[i] = new QMutex;
 	}
 

